gtsfm/frontend/correspondence_generator/image_correspondence_generator.py

```diff
@@ -28,42 +34,98 @@ def __init__(self, matcher: ImageMatcherBase, deduplicate: bool = True) -> None:
         """
         self._matcher = matcher
 
-        if deduplicate:
-            self._aggregator = KeypointAggregatorDedup()
-        else:
-            self._aggregator = KeypointAggregatorUnique()
+        self._aggregator: KeypointAggregatorBase = (
+            KeypointAggregatorDedup() if deduplicate else KeypointAggregatorUnique()
+        )
 
-    def create_computation_graph(
+    def apply(
```

Comment:
why are we doing two view estimation in the correpondence generator?

Reply:
Yeah this is something I did to just fit in the pattern. Maybe we need to rename `CorrespondenceGenerator`?